### PR TITLE
Migrate remaining containerd-build-* jobs to the community bucket and introduce postsubmit jobs

### PR DESCRIPTION
**Pull request by upodroid**
Follow up to #29986
Part of https://github.com/kubernetes/k8s.io/issues/1469
Part of https://github.com/kubernetes/test-infra/issues/29722
Fixes https://github.com/kubernetes/test-infra/issues/29995

I did a few changes:

- The build jobs now run as postsubmits. A build is triggered when a commit is merged on the relevant branches instead of building containerd every 30 minutes. (We can use that compute to test something more useful)
- The CI job runs once a day.
- I updated the OWNERS file in containerd folder to match the sig-node folder
- containerd jobs will be running on community infrastructure

### CODE REVIEW

**Comment by upodroid** [comment]
/cc @SergeyKanzhelev 

This is ready to be merged.

File: config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                memory: 6Gi
+              requests:
+                cpu: 4
+                memory: 6Gi
+      annotations:
+        testgrid-dashboards: sig-node-containerd,containerd-postsubmits
+        testgrid-tab-name: post-containerd-build-1.6
+        description: "builds release/1.6 branch of upstream containerd"
+
+    - name: post-containerd-build-1-7
```
Comment (akhilerm) [inline config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml:66]: question: Where do the logs get uploaded to? In the older jobs there is a [`--upload=gs://kubernetes-jenkins/logs`](https://github.com/kubernetes/test-infra/blob/423fd8bb893e8d0576964ce309480fd504021b32/config/jobs/kubernetes/sig-node/containerd.yaml#L82)

File: config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230616-e730b60769-master
+            command:
+              - runner.sh
+            args:
+              - test/build-test-images.sh
+            env:
+              - name: DEPLOY_DIR
+                value: main
+              - name: PROJECT
```
Comment (akhilerm) [inline config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml:115]:
```suggestion
              - name: DEPLOY_BUCKET
```

**Review by akhilerm — COMMENTED** [review]
Have a few comments on the changes:

Also 2 follow up changes need to be done:
1. The default bucket referred in [`containerd/containerd`](https://github.com/search?q=repo%3Acontainerd%2Fcontainerd%20cri-containerd-staging&type=code) should also be changed. 
2. References to `cri-containerd-staging` in the [env](https://github.com/search?q=repo%3Akubernetes%2Ftest-infra%20cri-containerd-staging&type=code) files used for deploying VMs with containerd should be changed, else those node-e2e jobs will start failing.

File: config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml (hunk 2 above)
Comment (upodroid) [reply config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml:115]: This script pushes images and not blobs. Have a look at https://github.com/containerd/containerd/blob/main/test/build-test-images.sh#L25 and the last 2 lines

**Comment by upodroid** [comment]
> * The default bucket referred in [`containerd/containerd`](https://github.com/search?q=repo%3Acontainerd%2Fcontainerd%20cri-containerd-staging&type=code) should also be changed.
> * References to `cri-containerd-staging` in the [env](https://github.com/search?q=repo%3Akubernetes%2Ftest-infra%20cri-containerd-staging&type=code) files used for deploying VMs with containerd should be changed, else those node-e2e jobs will start failing.

1. Not really required and might break other users
2. I will do this in a separate PR once this is merged. The old bucket will still be available but it will have stale containerd builds(by a few days which isn't critical).

Can I get a LGTM?

File: config/jobs/containerd/containerd/OWNERS
[the remark this answers is not in the document]
Comment (upodroid) [reply config/jobs/containerd/containerd/OWNERS:1]: done

File: config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml (hunk 2 above)
Comment (akhilerm) [reply config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml:115]: Then, shouldnt it be PROJECT in the periodics also? https://github.com/kubernetes/test-infra/pull/30025/files#diff-669927a638396eb51d888d0752bf589ea0ee622ac29fab684feda313eb88edd7R129

Comment (akhilerm) [reply config/jobs/containerd/containerd/containerd-postsubmit-jobs.yaml:115]: @upodroid Can you check this?, Shouldnt the postsubmit and periodics be the same.

File: config/jobs/containerd/containerd/containerd-periodic-jobs.yaml [hunk 3]
```diff
@@ -34,5 +34,111 @@ periodics:
         privileged: true
```
Comment (akhilerm) [inline config/jobs/containerd/containerd/containerd-periodic-jobs.yaml:34]: If `preset-dind-enabled: "true"`, this is removed. do we still need privileged?

Comment (upodroid) [reply config/jobs/containerd/containerd/containerd-periodic-jobs.yaml:34]: no, the job isn't building docker images

Comment (upodroid) [reply config/jobs/containerd/containerd/containerd-periodic-jobs.yaml]: Fixed

**Comment by akhilerm** [comment]
@upodroid I had one more question here https://github.com/kubernetes/test-infra/pull/30025#discussion_r1270250295

**Comment by upodroid** [comment]
As I said earlier, we can merge this PR now and I can open a separate one to adjust the containerd env files. I'll open it today for review. We'll have stale builds for containerd for a few hours/days after this is merged depending on how fast PRs are being merged in containerd.

**Review by mikebrow — COMMENTED** [review]
interesting

**Pull request merged**